### PR TITLE
Added groups to the call cross_val_score which will enable alternativ…

### PR DESCRIPTION
**Pull request by zoso95**
…e forms of cross validation.

## What does this PR do?
This PR addds a group parameter to fit, which gets passed down into cross_val_score. Having a groups, combined with a CV that isn't an integer let's users do stuff like GroupKFold cross validation.

For example, this code now works
```
tpot = TPOTClassifier(generations=5, population_size=20, verbosity=2, cv=GroupKFold(n_splits=2))
tpot.fit(X_train, y_train, groups=g)
```

## Where should the reviewer start?

It is a small PR, so it should be straightforward. 

## How should this PR be tested?
It should suffice to run the local unit tests. 

## What are the relevant issues?

https://github.com/rhiever/tpot/issues/312

## Screenshots (if appropriate)
N/A

## Questions:

- Do the docs need to be updated?
I updated them
- Does this PR add new (Python) dependencies?
No

### CODE REVIEW

**Comment by weixuanfu2016** [comment]
Looks good. Could you please merge it to dev branch instead of master branch? 

May also need update description of cv parameter in TPOTbase class in`base.py` and `driver.py`

**Comment by rhiever** [comment]
I changed the base to the development branch. Looks like there are several conflicts that need to be resolved now.

**Comment by zoso95** [comment]
Okay, it appears that all conflicts have been taken care of, and the tests pass. PTAL.

File: tpot/gp_deap.py [hunk 1]
```diff
@@ -360,7 +360,8 @@ def _wrapped_cross_val_score(sklearn_pipeline, features, classes,
         cv=cv,
         n_jobs=1,
         verbose=0,
-        fit_params=sample_weight_dict
+        fit_params=sample_weight_dict,
+      , groups=groups
```
Comment (weixuanfu2016) [inline tpot/gp_deap.py]: The comma in line 364 need to be cleaned.

File: docs_sources/using.md [hunk 2]
```diff
@@ -92,8 +92,8 @@ See the section on <a href="#scoringfunctions">scoring functions</a> for more de
 <tr>
 <td>-cv</td>
 <td>CV</td>
-<td>Any integer >1</td>
-<td>Number of folds to evaluate each pipeline over in k-fold cross-validation during the TPOT optimization process.</td>
+<td>Any integer >1 or cross-validation generator</td>
+<td>If CV is a number, then it is the number of folds to evaluate each pipeline over in k-fold cross-validation during the TPOT optimization process. If it is an object then it is an object to be used as a cross-validation generator.</td>
```
Comment (weixuanfu2016) [inline docs_sources/using.md]: I don't think it is will work in command-line mode. For interactive mode, please update:
Line 246 in `using.md`
Line 131 in `base.py`

Comment (zoso95) [inline docs_sources/using.md]: Ah, you're right. I realized that it wouldn't work in interactive mode, but I accidentally updated the wrong section.

**Comment by zoso95** [comment]
Done the changes you suggested.

**Comment by weixuanfu2016** [comment]
Thank you!  Looks very good!

**Comment by rhiever** [comment]
Will review when I get back to the office next week.

**Comment by rhiever** [comment]
Worked out some merge conflicts. @zoso95, can you please add some unit tests to the PR that test and demonstrate the intended functionality?

**Comment by zoso95** [comment]
Okay, I added a test.

**Comment by rhiever** [comment]
Thanks @zoso95 -- merged and will come out in the 0.8 release.

**Pull request merged**